### PR TITLE
Implement more ciphers in OpenSSH

### PR DESCRIPTION
**Pull request by Leo1003**
This PR adds implementation for the ciphers supported in OpenSSH.

Closes: #108

### CODE REVIEW

**Comment by williamdes** [comment]
Here is my public key using a Yubikey, is it implemented ?
```
sk-ecdsa-sha2-nistp256@openssh.com AAAAInNrLWVjZHNhLXNoYTItbmlzdHAyNTZAb3BlbnNzaC5jb20AAAAIbmlzdHAyNTYAAABBBHsGFeKSVIeBuAXGVxAyBqq5x55neL8R8XfQUrEGsUwNvf5xKyFwtM8hMY60TXqf/DtMMrGJl0eiAen4Lgs+BMwAAAAfc3NoOndpbGxpYW1kZXMreWJtLTIwMjNAd2Rlcy5mcg== williamdes+ybm-2023@wdes.fr
```

**Comment by tarcieri** [comment]
@williamdes this PR is adding symmetric ciphers for key encryption, not digital signature algorithms.

That said that key is using ECDSA/NIST P-256, which is supported by the latest release: https://docs.rs/ssh-key/latest/ssh_key/#supported-algorithms

**Comment by williamdes** [comment]
> @williamdes this PR is adding symmetric ciphers for key encryption, not digital signature algorithms.
> 
> That said that key is using ECDSA/NIST P-256, which is supported by the latest release: https://docs.rs/ssh-key/latest/ssh_key/#supported-algorithms

Thanks for your kind reply
if I can be of some use to test stuff let me know

**Comment by Leo1003** [comment]
I think this PR is ready for review.

This PR introduces three new features to control which cipher is supported:
- des
- chacha20poly1305
- aes-cbc

File: ssh-key/Cargo.toml [hunk 1]
```diff
@@ -60,16 +64,19 @@ std = [
...
 dsa = ["dep:bigint", "dep:dsa", "dep:sha1", "alloc", "signature/rand_core"]
 ecdsa = ["dep:sec1"]
 ed25519 = ["dep:ed25519-dalek", "rand_core"]
-encryption = [ "alloc", "dep:aes", "dep:bcrypt-pbkdf", "dep:ctr", "rand_core"]
+encryption = [ "alloc", "dep:aes", "dep:bcrypt-pbkdf", "dep:cbc", "dep:ctr", "rand_core"]
 getrandom = ["rand_core/getrandom"]
 p256 = ["dep:p256", "ecdsa"]
 p384 = ["dep:p384", "ecdsa"]
 rsa = ["dep:bigint", "dep:rsa", "alloc", "rand_core"]
+des = ["dep:des", "encryption"]
```
Comment (tarcieri) [inline ssh-key/Cargo.toml:79]: Please put this in alphabetical order

Comment (tarcieri) [reply ssh-key/Cargo.toml:79]: Also perhaps call it `3des` or `tdes` for clarity

Comment (Leo1003) [reply ssh-key/Cargo.toml]: Renamed to tdes

**Review by tarcieri — APPROVED** [review]
Looks good, thanks!

**Pull request merged**